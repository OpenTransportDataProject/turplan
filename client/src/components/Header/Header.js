import React, { Component } from "react";
import logo from "./images/route.png";
import settings from "./images/settings.png";

import {
  HeaderContainer,
  MainTitle,
  LogoImg,
  SettingsImg,
  PromoText,
  MapButton
} from "./HeaderStyles";

//<LogoImg src={logo} alt="logo" />
/* <SettingsImg
  src={settings}
  alt="settings"
  onClick={() => console.log("Klikket på innstillinger......")}
/>*/

class Header extends Component {
  render() {
    return (
      <HeaderContainer>
        <MainTitle>Turplan</MainTitle>
        <PromoText>
          <h2>Her kan du finne turer og parkering</h2>
          <p>Nå kan du lett finne turer i ditt nærområde og planlegge turen dit. Her kan du hente ut parkeringsplasser og ladestasjoner til elbil ved noen få tastetrykk.  </p>
          <MapButton onClick={this.props.swapComponent.bind(this)}>
            Gå til kart
          </MapButton>
        </PromoText>

      </HeaderContainer>
    );
  }
}

export default Header;
