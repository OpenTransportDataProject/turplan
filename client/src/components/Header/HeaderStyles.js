import styled from "styled-components";

export const HeaderContainer = styled.div`
  flex: 1;
  background-color: rgb(58, 77, 109,0.4);
  background-image:url("images/dd.jpg");
  background-size:cover;
`;

export const MainTitle = styled.h1`
  font-size: 50px;
  color: white;
  text-align:center;
  @media(max-width:768px){
    text-align:left;
    margin-left:10%;
  }
`;

export const PromoText = styled.div`
  color: white;
  text-align:left;
  margin-left:10%;
  margin-bottom:40px;
  max-width:50%;
  font-size:18px;
  @media(max-width:768px){
    max-width:80%;
  }
`;

export const LogoImg = styled.img`
  width: 50px;
  height: 50px;
`;

export const SettingsImg = styled.img`
  width: 32px;
  height: 32px;
`;

export const MapButton = styled.button`
  display: inline-block;
  padding: 20px;
  font-size: 1em;
  background-color: #ddd;
  border: none;
  border-radius: 5px;
  &:hover{
    cursor:pointer;
  }
`;
