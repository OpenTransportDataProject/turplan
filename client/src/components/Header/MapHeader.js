import React from "react";

export function MapHeader() {
  return <div>TopTrip</div>;
}

export default MapHeader;
