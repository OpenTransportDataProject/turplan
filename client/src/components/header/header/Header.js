
import React, { PureComponent } from 'react';
import Leaflet from 'leaflet';

class Header extends PureComponent {
  render() {
    return (
      <h1>Page Title</h1>
    );
  }
}

export default Header;
