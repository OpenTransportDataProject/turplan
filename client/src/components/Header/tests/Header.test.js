import React from 'react';
import Header from '../Header.js';
import renderer from 'react-test-renderer';

test('Renders correctly', () => {
  const mockFn = jest.fn();
  const component = renderer.create(
    <Header swapComponent={mockFn}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
