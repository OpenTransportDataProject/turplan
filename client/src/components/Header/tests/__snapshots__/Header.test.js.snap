// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="sc-bdVaJa ebqGbM"
>
  <h1
    className="sc-bwzfXH glACvo"
  >
    Turplan
  </h1>
  <div
    className="sc-htpNat cNrdsZ"
  >
    <h2>
      Her kan du finne turer og parkering
    </h2>
    <p>
      Nå kan du lett finne turer i ditt nærområde og planlegge turen dit. Her kan du hente ut parkeringsplasser og ladestasjoner til elbil ved noen få tastetrykk.
       
    </p>
    <button
      className="sc-bxivhb Dblxq"
      onClick={[Function]}
    >
      Gå til kart
    </button>
  </div>
</div>
`;
